modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/FDSViewFragment.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {FrontendDataSet} from '@liferay/frontend-data-set-web';
+
+import PropTypes from "prop-types";
+import React, {useEffect, useState} from "react";
+
+function FDSViewFragment({
```

Comment:
This is the component that preloads filter CXs before rendering the final FDS.
We could also use this intermediate component for preloading more things if needed...

Reply:
Well the idea sounds good to me, I would suggest using the same pattern for all the client extensions needed by this component. We already have cell renderers, and there are chances that more types will come. 

A centralized point to load all of them looks a good decision from an architectural point of view because:
 * We apply the same logic to load
 * Still we may decide to go on if something can't be loaded, providing sensible defaults. For example, the default cell renderer. And it's easier to log all this in the browser console, still providing a limited experience by rendering a correctly configured FDS
 * Both fragment (or any other client code instantiating FDS) and  the FDS itself don't have to worry about this anymore.

On the other side:
* The `useEffect()` forces a first render of the FDS with the initial state of `resolvedFilters` (null). This may be improved by setting an initial state with all filters which don't need `import()`. This way, the next render of the FDS will include all the other filters
* If we wait till everything is loaded, FDS rendering may be delayed. This can become significant in complex scenarios, where a more lazy approach could make sense. For this to happen, FDS should handle this loading and just render what's needed once the CX is imported. This makes things unnecesarily complex at this moment, so I'm fine with the "all-or-nothing" approach where the full FDS is rendered
* Still, I think this is more convenient if managed from the FDS itself as otherwise, the fragment entry point is the only one able to deal with client extension loading. This looks suboptimal to me because code outside the fragment may be interested in configuring an FDS instance with external URLs as well

Reply:
For 1 and 2, I understand it sounds better to render in two phases but that doesn't hold if there's preloaded data (which we don't allow yet for CXs). If you had preloaded data for a CX you would see a first render with some rows coming from a partial filtering, then a new render would trigger (perhaps after a noticeable amount of time if the network is slow and the CX is hosted outside of DXP) and the dataset would possibly show completely different rows.

Regarding point 3, I agree. The loading logic should be done in FDS, not the fragment.